### PR TITLE
Improve mssql backups

### PR DESCRIPTION
**Pull request by Mart124**
Hi,

This PR does several things :
- it modifies DB backup start time, so that they are named using the the day that has just gone ;
- it deletes old backups only if backup succeeded, for security reasons.

Additionally :
- it puts year first in backup timestamp ;
- it allows to pass the backup timestamp as a parameter, if script is launched manually for example ;
- it keeps a whole full month of backups.

Thank you !

### CODE REVIEW

**Comment by Mart124** [comment]
I may have found a better to check that backup is successful, before removing old ones.

File: util/MsSql/crontab [hunk 1]
```diff
@@ -1,3 +1,3 @@
-0 0 * * * bitwarden /backup-db.sh >> /var/log/cron.log 2>&1
+50 23 * * * bitwarden /backup-db.sh >> /var/log/cron.log 2>&1
```
Comment (kspearrin) [inline util/MsSql/crontab]: Could we change this to `59 23 * * *`? Why the 10 minute gap?

File: util/MsSql/backup-db.sh [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
-# Delete backup files older than 30 days
-find /etc/bitwarden/mssql/backups/ -type f -name '*.BAK' -mindepth 1 -mtime +30 -delete
+# Backup timestamp
+export now=${1:-$(date +%Y%m%d_%H%M%S)}
 
 # Do a new backup
 /opt/mssql-tools/bin/sqlcmd -S localhost -U sa -P ${SA_PASSWORD} -i /backup-db.sql
+
+# Delete backup files older than 30 days
+grep -B1 "BACKUP DATABASE successfully" /var/opt/mssql/log/errorlog | grep -q _$now.BAK &&
```
Comment (kspearrin) [inline util/MsSql/backup-db.sh]: What does this `-B1` option do?

File: util/MsSql/backup-db.sh [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+find /etc/bitwarden/mssql/backups/ -mindepth 1 -type f -name '*.BAK' -mtime +32 -delete
```
Comment (kspearrin) [inline util/MsSql/backup-db.sh]: Why +32 now?

File: util/MsSql/backup-db.sh [hunk 4]
```diff
@@ -1,7 +1,11 @@
 #!/bin/sh
 
-# Delete backup files older than 30 days
-find /etc/bitwarden/mssql/backups/ -type f -name '*.BAK' -mindepth 1 -mtime +30 -delete
+# Backup timestamp
+export now=${1:-$(date +%Y%m%d_%H%M%S)}
```
Comment (kspearrin) [inline util/MsSql/backup-db.sh]: Not sure why a user would every want to pass in a value. I would think it should always default to now date

File: util/MsSql/crontab (hunk 1 above)
Comment (Mart124) [reply util/MsSql/crontab]: You're right, no real reason, done 👍

File: util/MsSql/backup-db.sh (hunk 2 above)
Comment (Mart124) [reply util/MsSql/backup-db.sh]: `-B1` means one line back.
When a backup succeeds, log looks like this :
```
2018-05-29 06:03:58.53 Backup      Database backed up. Database: vault, creation date(time): 2018/03/05(08:26:55), pages dumped: 22615, first LSN: 70:17104:37, last LSN: 70:17136:1, number of dump devices: 1, device information: (FILE=1, TYPE=DISK: {'/etc/bitwarden/mssql/backups/vault_FULL_20180529_080349.BAK'}). This is an informational message only. No user action is required.
2018-05-29 06:03:58.57 Backup      BACKUP DATABASE successfully processed 22602 pages in 8.556 seconds (20.637 MB/sec).
```

When then look for `BACKUP DATABASE successfully` and the line before (`-B1`).
Finally, in these 2 lines, we look for the backup name (second `grep`).

File: util/MsSql/backup-db.sh (hunk 3 above)
Comment (Mart124) [reply util/MsSql/backup-db.sh]: To be sure to have a whole **full month** of backups.
Let's assume we are the 31st of May, you said you keep one month of backups, and someone (a manager, a coworker...) asks you to restore the database as it was on the 1st day of the month...
With 32 we're safe :)

**Comment by Mart124** [comment]
Thx for merging 👍

**Pull request merged**